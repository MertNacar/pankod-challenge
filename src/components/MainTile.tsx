import React from "react";

interface IProp {
  route: string;
  image: string;
  alt: string;
  title: string;
}

const SubNavBar: React.FC<IProp> = (props) => {
  return (
    <div className="card">
      <a href={props.route}>
        <img
          src={props.image}
          className="card-img-top"
          alt={props.alt}
          style={{ maxHeight: 220, maxWidth: 180 }}
        />
        <div className="card-img-overlay d-flex align-items-center justify-content-center">
          <h3 className="card-title text-light">{props.title}</h3>
        </div>
      </a>
    </div>
  );
};

export default SubNavBar;
