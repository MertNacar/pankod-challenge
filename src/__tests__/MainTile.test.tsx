import React from "react";
import { shallow } from "enzyme";
import MainTile from "../components/MainTile";

describe("MainTile Component", () => {
  const wrapper = shallow(
    <MainTile
      route="/series"
      image="https://streamcoimg-a.akamaihd.net/000/128/61/12861-PosterArt-ec32a81986a45eac7e080112075ab466.jpg"
      alt="Wolf Creek"
      title="Wolf Creek"
    />
  );

  it("is exists", () => {
    expect(wrapper.exists()).toBeTruthy();
  });

  it("return one MainTile component", () => {
    expect(wrapper).toMatchSnapshot("Content");
  });

  it("should match titles as Wolf Creek", () => {
    const title = wrapper.find(".text-light").text();
    expect(title).toBe("Wolf Creek");
  });
});
