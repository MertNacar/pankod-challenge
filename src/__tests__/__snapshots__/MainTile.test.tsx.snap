// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainTile Component return one MainTile component: Content 1`] = `
<div
  className="card mb-3"
  style={
    Object {
      "maxHeight": 220,
      "maxWidth": 180,
    }
  }
>
  <a
    href="/series"
  >
    <img
      alt="Wolf Creek"
      className="card-img-top"
      src="https://streamcoimg-a.akamaihd.net/000/128/61/12861-PosterArt-ec32a81986a45eac7e080112075ab466.jpg"
      style={
        Object {
          "maxHeight": 220,
          "maxWidth": 180,
        }
      }
    />
    <div
      className="card-img-overlay d-flex align-items-center justify-content-center"
    >
      <h3
        className="mytext card-title text-light"
      >
        Wolf Creek
      </h3>
    </div>
  </a>
</div>
`;
